src/HttpClients/TestGuzzleHttpClient.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Telegram\Bot\HttpClients;
+
+class TestGuzzleHttpClient extends GuzzleHttpClient
+{
+    /** @var  int */
+    protected $timeOut;
+    /** @var  int */
+    protected $connectTimeOut;
+
+    public function send(
+        $url,
+        $method,
+        array $headers = [],
+        array $options = [],
+        $timeOut = 30,
+        $isAsyncRequest = false,
+        $connectTimeOut = 10
+    ) {
+        $this->timeOut = $timeOut;
+        $this->connectTimeOut = $connectTimeOut;
+
+        return parent::send(
+            $url,
+            $method,
+            $headers,
+            $options,
+            $timeOut,
+            $isAsyncRequest,
+            $connectTimeOut
+        );
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getTimeOut()
+    {
+        return $this->timeOut;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getConnectTimeOut()
+    {
+        return $this->connectTimeOut;
+    }
+
+}
```

Comment:
OK, I'll do everything a little bit later. Thank you for review!
On 09-Dec-2015 8:25 am, "Syed Irfaq R." <notifications@github.com> wrote:

> In src/HttpClients/TestGuzzleHttpClient.php
> <https://github.com/irazasyed/telegram-bot-sdk/pull/80#discussion_r47041091>
> :
>
> > +     * @return int|null
> > +     */
> > +    public function getTimeOut()
> > +    {
> > +        return $this->timeOut;
> > +    }
> > +
> > +    /**
> > +     * @return int|null
> > +     */
> > +    public function getConnectTimeOut()
> > +    {
> > +        return $this->connectTimeOut;
> > +    }
> > +
> > +}
>
> Nevermind, Just saw what you did. Please move these methods to the parent
> class itself.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/irazasyed/telegram-bot-sdk/pull/80/files#r47041091>.
>